static/js/import_wizard2/components/mapping_page.tsx

```diff
@@ -55,9 +56,32 @@ export function MappingPage(props: MappingPageProps): JSX.Element {
   }
 
   useEffect(() => {
-    // TODO(beets): Use server-side detection API.
-    const predictedMapping = getPredictions(props.csvData, placeDetector);
-    setPredictedMapping(predictedMapping);
+    /* Build JSON of column data for detection API Request */
+    const column_ids = new Map(
+      props.csvData.orderedColumns.map((column) => {
+        return [column.columnIdx, column.id];
+      })
+    );
+    const column_headers = new Map(
+      props.csvData.orderedColumns.map((column) => {
+        return [column.columnIdx, column.header];
+      })
+    );
+    const columnParameters = new Map<string, unknown>([
+      ["column_ids", column_ids],
+      ["column_headers", column_headers],
+      ["column_values", props.csvData.columnValuesSampled],
+    ]);
+    /* Use server-side API for column detection */
+    axios
+      .post(`/api/detection/detect_columns`, JSON.stringify(columnParameters))
+      .then((resp) => {
+        const parsedResponse = parseDetectionApiResponse(resp.data);
+        setPredictedMapping(parsedResponse);
+      })
+      .catch((error) => {
+        console.log(error);
```

Comment:
maybe set predictedMapping to an empty mapping (i.e., `new Map()`)

Reply:
added!